tests/backup/backup_helper.go

```diff
@@ -5434,3 +5434,128 @@ func RestartAllVMsInNamespace(namespace string, waitForCompletion bool) error {
+	if err != nil {
+		return err
+	}
+	if currentKubevirtVersionSemVer.GreaterThanOrEqual(versionToUpgradeSemVer) {
+		return fmt.Errorf("kubevirt cannot be upgraded from [%s] to [%s]", currentKubevirtVersionSemVer.String(), versionToUpgradeSemVer.String())
+	} else {
+		log.InfoD("Upgrade path chosen (%s) ---> (%s)", current, versionToUpgrade)
+	}
+
+	// Generating the manifest URL and applying it to begin upgrade
+	manifestYamlURL := fmt.Sprintf("https://github.com/kubevirt/kubevirt/releases/download/%s/kubevirt-operator.yaml", versionToUpgrade)
+	_, err = kubectlExec([]string{fmt.Sprintf("--kubeconfig=%v", CurrentClusterConfigPath), "apply", "-f", manifestYamlURL})
+	if err != nil {
+		return err
+	}
+
+	// Waiting till the upgrade is complete
+	t := func() (interface{}, bool, error) {
+		upgradedVersion, err := k8sKubevirt.GetVersion()
+		if err != nil {
+			return "", false, err
+		}
+		if upgradedVersion != versionToUpgrade {
+			return "", true, fmt.Errorf("waiting for kubevirt control plane to be upgraded to [%s] but got [%s]", versionToUpgrade, upgradedVersion)
+		}
+		return "", false, nil
+	}
+	_, err = task.DoRetryWithTimeout(t, 10*time.Minute, 30*time.Second)
+	if err != nil {
+		return err
+	}
+
+	// Waiting for all pods in kubevirt namespace to be Ready
+	kubevirtCheck := func() (interface{}, bool, error) {
+		pods, err := k8sCore.GetPods(KubevirtNamespace, make(map[string]string))
+		if err != nil {
+			return "", false, err
+		}
+		allReady := true
+		for _, p := range pods.Items {
+			log.Infof("Checking status for pod - %s", p.GetName())
+			for _, condition := range p.Status.Conditions {
+				if condition.Type == corev1.PodReady {
+					log.Infof("Pod [%s] in [%s] namespace is in [%s] state", p.GetName(), KubevirtNamespace, condition.Type)
+					allReady = true
+					break
+				} else {
+					log.Infof("Pod [%s] in [%s] namespace expected state - [%s], but got [%s]", p.GetName(), KubevirtNamespace, corev1.PodReady, condition.Type)
+					allReady = false
+				}
+			}
+			if !allReady {
+				return "", true, fmt.Errorf("waiting for all the pods in %s namespace to be ready", KubevirtNamespace)
+			}
+		}
+		return "", false, nil
+	}
+	_, err = task.DoRetryWithTimeout(kubevirtCheck, 10*time.Minute, 30*time.Second)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Kubevirt control plane upgraded from [%s] to [%s]", current, versionToUpgrade)
+
+	// Workload upgrade
+	if workloadUpgrade {
+		patchString := `[
+  		{"op": "replace", "path": "/spec/imagePullPolicy", "value": "IfNotPresent"},
+  		{"op": "replace", "path": "/spec/workloadUpdateStrategy", "value": {"workloadUpdateMethods": ["Evict"], "batchEvictionSize": 10, "batchEvictionInterval": "1m"}}
+	]`
+
+		_, err = kubectlExec([]string{fmt.Sprintf("--kubeconfig=%v", CurrentClusterConfigPath), "patch", "kubevirt", "kubevirt", "-n", "kubevirt", "--type=json", fmt.Sprintf("-p=%s", patchString)})
```

Comment:
We have already. vendored in kubevirt also for version.. Could you just do Object update instead of kubectl patch?